src/realm/object-store/sync/sync_session.cpp

```diff
@@ -637,6 +627,29 @@ void SyncSession::handle_error(SyncError error)
                         return; // do not propgate the error to the user at this point
                 }
                 break;
+            case sync::ProtocolErrorInfo::Action::MigrateToFLX:
+                if (!m_original_sync_config->flx_sync_requested) {
+                    m_migration_store->migrate_to_flx(error.migration_query_string);
+                    // restart session without notifying user
+                    next_state = NextStateAfterError::restart;
```

Comment:
I don't think we need this since we'll call `download_fresh_realm` similar to how client reset is handled above.